### ISSUE TITLE
5.1RC1: exclude from navigation ignored? registry.xml entries replacing a configuration instead of adding to an existing configuration.

### ISSUE THREAD

**Issue by zopyx** [plone/Products.CMFPlone#2171]
I run plone-demo.info.

All auto-generated folders does not show up in the tabs.

They also do not show up after switching the exclude from navigation on/off manually.

A newsly created "Folder" also does not show up in the tabs.

This can be reproduced on plone-demo.info (admin/admin).

<img width="787" alt="bildschirmfoto 2017-10-09 um 20 53 05" src="https://user-images.githubusercontent.com/594239/31353772-1e978bda-ad34-11e7-8590-e6be3ce4be9d.png">

<img width="949" alt="bildschirmfoto 2017-10-09 um 20 52 51" src="https://user-images.githubusercontent.com/594239/31353771-1e967d3a-ad34-11e7-8fb8-18339219642c.png">

**Comment by pbauer** [comment]
I cannot reproduce that issue. demo.plone.de is also running 5.1rc1 and doe not have it.

**Comment by zopyx** [comment]
Weird.

**Comment by zopyx** [comment]
I tracked this down to my  `registry.xml`:

https://bitbucket.org/ajung/zopyx.plone.smashdocs/src/13111287936476d0b80b84450194e2be0abfad6c/zopyx/plone/smashdocs/profiles/default/registry.xml?at=master&fileviewer=file-view-default

In Plone 5.0 the two content types where added to the registry, in Plone 5.1 the two content types seem to replace existing entries. Although there is no `prune="True"` included with the configuration. 

In general the behaviour should and must remain additive (unless explicitly stated through `prune="True"`. Otherwise it is impossible to write proper `add`-on configurations.

**Comment by zopyx** [comment]
Setting `purge="false"` in my registry.xml resolved the issue

https://bitbucket.org/ajung/zopyx.plone.smashdocs/commits/e36db0e93eea252f4125b6030c5af39b9fd88af7

However I still believe that the change between 5.0 and 5.1 is a backward compatibility foul.

**Comment by thet** [comment]
@zopyx it`s ``purge="False"`` what is needed, otherwise existing entries will be cleared and overwritten.

You have the ``plone.excludefromnavigation`` ( https://github.com/plone/plone.app.dexterity/blob/master/plone/app/dexterity/behaviors/exclfromnav.py ) behavior  added to your content type?

**Comment by zopyx** [comment]
I am using this behavior for one of the two content-types in my package

https://bitbucket.org/ajung/zopyx.plone.smashdocs/src/e36db0e93eea252f4125b6030c5af39b9fd88af7/zopyx/plone/smashdocs/profiles/default/types/SmashFolder.xml?at=master&fileviewer=file-view-default#SmashFolder.xml-27

But how is this related? 

My point is that adding values to a registry entry should be really added instead of replacing existing entries. The default behavior seems to have changed between 5.0 and 5.1 in my opinion.

**Comment by thet** [comment]
AFAIK ``purge="False"`` way was always the same with Generic Setup. Although in the end it depends on the exporthandlers if they support the ``purge`` attribute and if they append or replace existing values. And that is indeed not totally consistent.

The link to your repo gives a access denied error.

**Comment by jensens** [comment]
Since we're using this in ``profiles/default/types/*.xml`` files quite often, I just looked up how it worked the last years with behaviors.

Given a ``Link.xml`` overriding the behaviors of the default Link (in order to reduce fields), I define a block like this

```
  <!-- Enabled behaviors -->
  <property name="behaviors">
    <element value="plone.dublincore" />
    <element value="plone.namefromtitle" />
    <element value="plone.allowdiscussion" />
    <element value="plone.excludefromnavigation"/>
    <element value="plone.shortname" />
  </property>
```

Then the already existing behaviors are overridden. Only those defined here are active.

If I change this to:

```
  <!-- Enabled behaviors -->
  <property name="behaviors" purge="false">
    <element value="plone.dublincore" />
    ...
  />
  </property>
```

Then the behaviors are added to the already existing ones.

I do not see any difference in behavior between 4.3-latest, 5.0-latest, and 5.1rc1.

Overall I don't get what you mean by ``registry.xml`` here - overall the private links you gave are referencing FTI - a different story.

**Comment by zopyx** [comment]
Sorry but I did not notice that the repo is private (public now).

The excludefromnavigation behavior is unrelated.

In my original code for Plone 5.0 I had to this here in order to display my two content-types in portal_tabs:

https://bitbucket.org/ajung/zopyx.plone.smashdocs/src/13111287936476d0b80b84450194e2be0abfad6c/zopyx/plone/smashdocs/profiles/default/registry.xml?at=master&fileviewer=file-view-default#registry.xml-17:20

With Plone 5.1 this configuration led to the situation as seen in the screenshots: all standard content-types no longer show up in portal_tabs except the two specified.

So in Plone 5.0 both content-types have been added to the `plone.app.vocabularies.ReallyUserFriendlyTypes` vocabulary but in Plone 5.1 both types replaced the existing vocabulary terms. So the default behaviour in Plone 5.1 changed from `purge=False` to `purge=True`.

**Comment by jensens** [comment]
I don't think the default behavior changed. I have several projects coming from 4.3/5.0 with a whole bunch of registry.xml and had not a single problem b/c of this.

**Comment by ale-rt** [comment]
It seems that the setting exclude from navigation was ignored since a long time and it is now fixed on 5.2: see https://github.com/plone/Products.CMFPlone/pull/2743

I added an issue to backport the fix https://github.com/plone/Products.CMFPlone/issues/2758

**Comment by jensens** [comment]
I close the issue, because it addresses a Plone version that is no longer supported. 
If you think this is wrong please reopen the issue and assign a matching milestone.
